indico/modules/events/editing/client/js/components/FileManager/reducer.js

```diff
@@ -0,0 +1,110 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import _ from 'lodash';
+import {combineReducers} from 'redux';
+import * as actions from './actions';
+
+export default combineReducers({
+  fileTypes: (state = [], action) => {
+    const fileType = _.find(state, {id: action.fileTypeId});
+    switch (action.type) {
+      case actions.MARK_UPLOADED:
+        action.file.state = 'added';
```

Comment:
I am wondering whether we should directly update the `file` object like that. It could probably have some side effects.

```javascript
const file = {...action.file, state: 'added'}l
...
    fileType.files = [...fileType.files, file];
} else {
    fileType.files = [file];

Reply:
It's not very likely, since the action object is not reused AFAIK. But it's probably nicer to omit that line and do:
```js
        if (fileType.allowMultipleFiles) {
          fileType.files = fileType.files.concat({...action.file, state: 'added'});
        } else {
```
What do you guys think?

Reply:
@mic4ael your version is also fine, of course...

Reply:
Throwing another version in the ring ;)

```js
fileTypes.files = [...fileType.files, {...action.file, state: 'added'}];
```